### PR TITLE
feat: add new test workflow

### PR DESCRIPTION
**Pull request by dejanzele**
Fixes #

#### Special notes for your reviewer:

┆Issue is synchronized with this [Jira Task](https://gr-oss.atlassian.net/browse/BATCH-391) by [Unito](https://www.unito.io)

### CODE REVIEW

**Review by pavlovic-ivan — DISMISSED** [review]
LGTM

**Comment by kannon92** [comment]
Our required workflows are looking for `go-lint (1.20)` and `go-unit-tests (1.20)`.  Otherwise I think this LGTM.

**Review by kannon92 — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+on:
+  push:
+    branches-ignore:
+      - gh-pages
+  pull_request:
+    branches-ignore:
+      - gh-pages
+  schedule:
+    # Run daily at 01:34, so we get notified if CI is broken before a pull request
```
Comment (Sharpz7) [inline .github/workflows/ci.yml:11]: Is there a reason for this exact time?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+      - name: Upload Test Reports Artifacts
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: ts-unit-test-reports
+          path: ./internal/lookout/ui/junit.xml
+          if-no-files-found: error
+
+      - name: Send Coverage Report to Codecov
```
Comment (Sharpz7) [inline .github/workflows/test.yml:49]: Are we able to test that this produces the results we expect?

File: makefile [hunk 3]
```diff
@@ -402,14 +402,7 @@ tests: gotestsum
...
 	function tearDown { docker rm -f redis postgres; }; trap tearDown EXIT
-	$(GOTESTSUM) -- $(shell go list ./internal/... | grep -v 'jobservice/repository') \
-		-coverprofile internal_coverage.xml -v  2>&1 | tee test_reports/internal.txt
-	env JSDBTYPE=sqlite $(GOTESTSUM) -- -v \
-			 ./internal/jobservice/repository/... 2>&1 | tee -a test_reports/internal.txt
-	env JSDBTYPE=postgres $(GOTESTSUM) -- -v \
-			 ./internal/jobservice/repository/... 2>&1 | tee -a test_reports/internal.txt
-	$(GOTESTSUM) -- -coverprofile pkg_coverage.xml -v ./pkg... 2>&1 | tee test_reports/pkg.txt
-	$(GOTESTSUM) -- -coverprofile cmd_coverage.xml -v ./cmd... 2>&1 | tee test_reports/cmd.txt
+	$(GOTESTSUM) --format short-verbose --junitfile test-reports/unit-tests.xml --jsonfile test-reports/unit-tests.json -- -coverprofile=test-reports/coverage.out -covermode=atomic ./cmd/... ./pkg/... $(go list ./internal/... | grep -v 'jobservice/repository')
```
Comment (Sharpz7) [inline makefile:405]: Can we get this moved to mage? Even if its just a simple command that is run?

**Comment by Sharpz7** [comment]
Have the intergration tests been changed in any way?

I had it run consistently at 6-7 mins, it seems to be taking longer than that.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dejanzele) [reply .github/workflows/ci.yml:11]: Nope, random

File: .github/workflows/test.yml (hunk 2 above)
Comment (dejanzele) [reply .github/workflows/test.yml:49]: https://app.codecov.io/gh/armadaproject/armada

**Comment by dejanzele** [comment]
not in this PR

**Review by severinson — APPROVED** [review]

**Pull request merged**